pkg/skaffold/docker/debugger/transform.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debugger
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/volume"
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug/types"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+var (
+	shouldTransform    bool
+	SupportVolumeMount = volume.VolumeCreateBody{Name: debug.DebuggingSupportFilesVolume}
+	TransformImage     = transformImage // For testing
+)
+
+type Transform func(string) string
+
+func EnableTransforms() {
+	shouldTransform = true
+}
+
+func transformImage(ctx context.Context, artifact graph.Artifact, cfg *container.Config, insecureRegistries map[string]bool, debugHelpersRegistry string) (map[string]types.ContainerDebugConfiguration, []*container.Config, error) {
+	if !shouldTransform {
```

Comment:
I strongly dislike global state being used like this.  It makes it very hard to reason about other code that may call into here.

Reply:
I agree, but as per my comment above I was mimicking the existing design for the k8s transforms. I'm going to add a TODO to remove the global state in favor of tracking the active command at runtime

Reply:
See my comment about moving this check into `pkg/skaffold/deploy/docker/deploy.Deployer.deploy()`.